network-api/networkapi/wagtailpages/factory/publication.py

```diff
@@ -63,6 +63,9 @@ class PublicationPageFactory(PageFactory):
     hero_image = SubFactory(ImageFactory)
     publication_file = DocumentFactory()
 
+    if random() > 0.5:
+        toc_thumbnail_image = SubFactory(ImageFactory)
```

Comment:
Ahh I see what you're doing here. For this we need a `@post_generation` method. 

First, import `post_generation`:
```python
from factory import post_generation
```

I also used ImageFactory from somewhere else:
```python
from networkapi.wagtailpages.factory.image_factory import ImageFactory
```
Then replace these two lines with this:
```suggestion
    @post_generation
    def toc_thumbnail_image(self, create, extracted, **kwargs):
        if random() < 0.5:
            self.toc_thumbnail_image = ImageFactory()
```

You should be able to copy and paste this to the ArticlePage factory down below as well.

Reply:
Wooohoo! Thanks! This def solves the problem I was having!